go/teams/service_helper.go

```diff
@@ -1582,10 +1582,24 @@ func FindNextMerkleRootAfterRemoval(mctx libkb.MetaContext, arg keybase1.FindNex
 	}
 
 	uv := vers.ToUserVersion()
-	logPoint := team.chain().GetLastUserLogPointWithPredicate(uv, func(p keybase1.UserLogPoint) bool {
-		return p.Role == keybase1.TeamRole_NONE || p.Role == keybase1.TeamRole_READER
-	})
-	if logPoint == nil {
+	logPoints := team.chain().inner.UserLog[uv]
+	demotionPredicate := func(p keybase1.UserLogPoint) bool {
+		if arg.AnyRoleAllowed {
+			return !p.Role.IsReaderOrAbove()
+		}
+		return !p.Role.IsWriterOrAbove()
+	}
+	var earliestDemotion int
+	var logPoint keybase1.UserLogPoint
+	for i := len(logPoints) - 1; i >= 0; i-- {
+		if demotionPredicate(logPoints[i]) {
+			earliestDemotion = i
+		} else if earliestDemotion != 0 {
+			logPoint = logPoints[earliestDemotion].DeepCopy()
+			break
+		}
+	}
+	if &logPoint == nil {
```

Comment:
oh i didn't see it before, this will never return false.

Reply:
huh. that's a little tricky, but i guess it makes sense. looks like i can't compare logPoint with an empty keybase1.UserLogPoint (because `struct containing keybase1.SignatureMetadata cannot be compared`) so im going to keep it as a pointer the whole time. is there another way to do this?

Reply:
In somecases, we can put an `IsNil()` around these XDR structs, to distinguish between an unset variable and one that's been populated. In this case, we likely don't have that, so a pointer is the better option.

Reply:
code fix and test are imminent

Reply:
cool. thanks.